### PR TITLE
AMP-78613 - Update Android migration guide

### PR DESCRIPTION
**Pull request by Mercy811**
# Amplitude Developer Docs PR

## Description

https://pr-843.d19s7xzcva2mw3.amplifyapp.com/data/sdks/android-kotlin/migration/

## Deadline

When do these changes need to be live on the site?

## Change type

- [ ] Doc bug fix. Fixes #[insert issue number]. Amplitude contributors include Jira issue number. 
- [ ] Doc update.
- [ ] New documentation.
- [ ] Non-documentation related fix or update.

# PR checklist:

- [ ] My documentation follows the style guidelines of this project.
- [ ] I previewed my documentation on a local server using `mkdocs serve`.
- [ ] Running `mkdocs serve` didn't generate any failures.
- [ ] I have performed a self-review of my own documentation.

@amplitude-dev-docs

### CODE REVIEW

File: docs/data/sdks/android-kotlin/migration.md [hunk 1]
```diff
@@ -1,8 +1,242 @@
...
+| `sessionTimeoutMillis` | Not supported |
+| `minTimeBetweenSessionsMillis` | `minTimeBetweenSessionsMillis` |
+| `serverUrl` | `serverUrl` defaults to `https://api2.amplitude.com/2/httpapi` while the maintenance SDK defaults to `https://api2.amplitude.com/` |
+| `useDynamicConfig` |  Not supported |
+
+### Send events
+
+#### `logEvent()`
+
+The `logEvent()` API maps to `track()`. the `eventProperties` is `JSONObject` type in the maintenance SDk while it's `Map<String, Any?>` type in the latest SDK.
```
Comment (justin-fiedler) [inline docs/data/sdks/android-kotlin/migration.md:90]:
```suggestion
The `logEvent()` API maps to `track()`. The `eventProperties` is `JSONObject` type in the maintenance SDK while it's `Map<String, Any?>` type in the latest SDK. 
```

**Review by justin-fiedler — APPROVED** [review]
Thanks @Mercy811

File: docs/data/sdks/android-kotlin/migration.md [hunk 2]
```diff
@@ -1,8 +1,242 @@
...
+| `eventUploadPeriodMillis` | `flushIntervalMillis` |
+| `eventUploadThreshold` | `flushQueueSize` |
+| `eventUploadMaxBatchSize` | Not supported |
+| `eventMaxCount` | Not supported |
+| `identifyBatchIntervalMillis` | `identifyBatchIntervalMillis` |
+| `flushEventsOnClose` | `flushEventsOnClose` |
+| `optOut` | `optOut` |
+| `trackingSessionEvents` | `trackingSessionEvents` |
+| `sessionTimeoutMillis` | Not supported |
+| `minTimeBetweenSessionsMillis` | `minTimeBetweenSessionsMillis` |
```
Comment (Mercy811) [inline docs/data/sdks/android-kotlin/migration.md:82]: Still working on these two configs and trying to explain them clearly.

- legacy Android SDK has two configs based on whether foreground session tracking is enabled or not
image

- latest Android SDK only and by default supports foreground session tracking with config `minTimeBetweenSessionsMillis`

File: docs/data/sdks/android-kotlin/migration.md [hunk 3]
```diff
@@ -1,8 +1,242 @@
...
+
+The revenue verification logic is on Amplitude's backend. So revenue verification still works after migrating to the latest Android SDK.
+
+## Advanced topics
+
+Most of the behaviours of the latest SDK remain the same with the maintenance SDK. Refer to the advanced topics sections of the [maintenance SDK](/data/sdks/android/#advanced-topics) and the [latest SDK](/data/sdks/android-kotlin/#advanced-topics) to learn more about a specific advanced topic.
+
 ## Comparison 
 
 --8<-- "includes/sdk-migration/sdk-migration-note.md"
```
[the remark this answers is not in the document]
Comment (Mercy811) [reply docs/data/sdks/android-kotlin/migration.md:242]: Thanks for point it out. Fixed

File: docs/data/sdks/android-kotlin/migration.md (hunk 2 above)
Comment (Mercy811) [reply docs/data/sdks/android-kotlin/migration.md]: The description of these two configs of legacy Android should be swapped and fixed it in another PR https://github.com/Amplitude-Developer-Docs/amplitude-dev-center/pull/848

File: docs/data/sdks/android-kotlin/migration.md [hunk 4]
```diff
@@ -1,13 +1,247 @@
...
+To migrate to `com.amplitude:analytics-android`, update your dependencies and instrumentation.
+
+## Terminology
+
+* `com.amplitude:android-sdk`: Maintenance Android SDK
+* `com.amplitude:analytics-android`: Latest Android SDK
+
+## Dependency
+
+Update build.gradle to remove the maintennace Android SDK and add the latest Android SDK. Then sync project with Gradle files.
```
Comment (yuhao900914) [inline docs/data/sdks/android-kotlin/migration.md:17]:
```suggestion
Update build.gradle to remove the maintenance Android SDK and add the latest Android SDK. Then sync project with Gradle files.
```

File: docs/data/sdks/android-kotlin/migration.md [hunk 5]
```diff
@@ -1,13 +1,247 @@
...
+| `eventUploadPeriodMillis` | `flushIntervalMillis` |
+| `eventUploadThreshold` | `flushQueueSize` |
+| `eventUploadMaxBatchSize` | Not supported |
+| `eventMaxCount` | Not supported |
+| `identifyBatchIntervalMillis` | `identifyBatchIntervalMillis` |
+| `flushEventsOnClose` | `flushEventsOnClose` |
+| `optOut` | `optOut` |
+| `trackingSessionEvents` | `trackingSessionEvents` |
+| `sessionTimeoutMillis` | Not supported |
+| `minTimeBetweenSessionsMillis` | `minTimeBetweenSessionsMillis`. The maintenance SDK disables foreground tracking by default and uses `sessionTimeoutMillis` when foreground tracking is disabled. `minTimeBetweenSessionsMillis` is used when foreground tracking is enabled by `enableForegroundTracking()`. However, the latest SDK enables foreground tracking and it's not configurable.|
```
Comment (yuhao900914) [inline docs/data/sdks/android-kotlin/migration.md:82]: I feel the content of `sessionTimeoutMillis` is more suitable to the `sessionTimeoutMillis` section. wdyt?
```suggestion
| `sessionTimeoutMillis` | Not supported.  The maintenance SDK disables foreground tracking by default and uses `sessionTimeoutMillis` when foreground tracking is disabled. However, the latest SDK enables foreground tracking and it's not configurable. |
| `minTimeBetweenSessionsMillis` | `minTimeBetweenSessionsMillis`|
```

File: docs/data/sdks/android-kotlin/migration.md [hunk 6]
```diff
@@ -1,13 +1,247 @@
...
+| `serverUrl` | `serverUrl` defaults to `https://api2.amplitude.com/2/httpapi` while the maintenance SDK defaults to `https://api2.amplitude.com/` |
+| `useDynamicConfig` |  Not supported |
+
+### Send events
+
+#### `logEvent()`
+
+The `logEvent()` API maps to `track()`. The `eventProperties` is `JSONObject` type in the maintenance SDK while it's `Map<String, Any?>` type in the latest SDK.
```
Comment (yuhao900914) [inline docs/data/sdks/android-kotlin/migration.md:90]: Additionally, we also support passing the event class directly into the track method. 
https://github.com/amplitude/Amplitude-Kotlin/blob/main/core/src/main/java/com/amplitude/core/Amplitude.kt#L135

File: docs/data/sdks/android-kotlin/migration.md [hunk 7]
```diff
@@ -1,13 +1,247 @@
...
+    ```diff
+    Revenue revenue = new Revenue()
+    revenue.setProductId("com.company.productId");
+    revenue.setPrice(3);
+    revenue.setQuantity(2);
+    - client.logRevenueV2(revenue);
+    + client.revenue(revenue);
+    ```
+
+#### Revenue verification
```
Comment (yuhao900914) [inline docs/data/sdks/android-kotlin/migration.md:232]: Thanks ❤️

File: docs/data/sdks/android-kotlin/migration.md (hunk 5 above)
Comment (Mercy811) [reply docs/data/sdks/android-kotlin/migration.md]: Yes! it's more clear

File: docs/data/sdks/android-kotlin/migration.md (hunk 7 above)
Comment (Mercy811) [reply docs/data/sdks/android-kotlin/migration.md:232]: No problem. Thanks for confirming with data pipeline team.

File: docs/data/sdks/android-kotlin/migration.md (hunk 6 above)
Comment (Mercy811) [reply docs/data/sdks/android-kotlin/migration.md:90]: Yes, but maintenance Android doesn't support that. Feel like we should only put equivalent methods of the maintenance SDK to minimize the page length. wdyt?

Comment (yuhao900914) [reply docs/data/sdks/android-kotlin/migration.md:90]: I'm fine with it.

**Review by yuhao900914 — APPROVED** [review]

**Pull request merged**